azext_connect/custom.py

```diff
@@ -499,6 +500,31 @@ def bind_webapp(
         sys.exit(1)
 
 
+def bind_webapp_postgres(
+    cmd, resource_group, name, appname, server, database,
+    username=None, password=None, authtype='MSI', permission=None
+):
+    try:
+        if authtype == 'Secret':
+            if not username:
+                username = prompt('Username: ')
+            if not password:
+                password = prompt_pass(msg='Password: ')
+        subscription = get_subscription_id(cmd.cli_ctx)
+        scope = '/subscriptions/{0}/resourceGroups/{1}'.format(subscription, resource_group)
+        source = '{0}/providers/Microsoft.Web/sites/{1}'.format(scope, appname)
+        target = _get_target_id(scope, postgres=server, database=database)
+        result = _bind(
+            cmd, subscription, resource_group, name, source,
+            target, authtype, permission, None, None, username, password
```

Comment:
Client_id and Client_secret are for Service Principal AuthType. It seems that service principal should be supported according to _params.py, thus it may be better not to set them as None.

Reply:
Have we already supported SP in the backend code?

Reply:
No, currently we have not support SP yet. I mean the design of cli seems take SP into consideration, and it may be better to follow the pattern.

Reply:
Make sense, let me add these 2 parameters.